crates/net/src/http/query.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt;
+use wasm_bindgen::UnwrapThrowExt;
+
+/// A sequence of URL query parameters, wrapping [`web_sys::UrlSearchParams`].
+pub struct QueryParams {
+    raw: web_sys::UrlSearchParams,
+}
+
+impl Default for QueryParams {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl QueryParams {
+    /// Create a new empty query parameters object.
+    pub fn new() -> Self {
+        // pretty sure this will never throw.
+        Self {
+            raw: web_sys::UrlSearchParams::new().unwrap_throw(),
+        }
+    }
+
+    /// Append a parameter to the query string.
+    pub fn append(&self, name: &str, value: &str) {
+        self.raw.append(name, value)
+    }
+}
+
+impl fmt::Display for QueryParams {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.raw.to_string())
+    }
+}
```

Comment:
What is the output of this? This seems like what should be a Debug implementation, not Display implementation

Reply:
Its the query string without `?`, so for instance `a=1&b=2`.

Reply:
If you can add a more readable debug implementation, that would be great. Something like the output of `Vec<(&str, &str)>` should be good